src/watcher/garbageCollector.ts

```diff
@@ -23,52 +24,74 @@ export class AppointmentStoreGarbageCollector {
         private readonly appointmentSubscriber: AppointmentSubscriber
     ) {}
 
+    // only allow the gc to be started once
+    private started: boolean = false;
+    private listener = (blockNumber: number) => this.removeExpired(blockNumber);
+    // only allow one collection at a time
+    private collecting = false;
+    // we want to record how many consecutive errors have taken place in gc
+    private consecutiveErrors = 0;
+
     /**
      * Start the monitoring for expired appointments
      */
     public start() {
-        this.provider.on("block", (blockNumber: number) => this.removeExpired(blockNumber));
+        if (!this.started) {
+            this.provider.on("block", this.listener);
+            this.started = true;
+        }
     }
 
-    private collecting = false;
+    /**
+     * Stop monitoring for expired events
+     */
+    public stop() {
+        if (this.started) {
+            this.started = false;
+            this.provider.removeListener("block", this.listener);
+        }
+    }
```

Comment:
Should we throw an error if start is called when it's already started, or stop when not started? Isn't it likely a bug if that happens?

Reply:
cool, I'll update

Reply:
I'll throw error for "start" but not for "stop" as errors in destructors can be dangerous. But I'll log error there.